libkbfs/backpressure_disk_limiter_test.go

```diff
@@ -142,6 +142,351 @@ func TestBackpressureTrackerCounters(t *testing.T) {
+	require.Equal(t, int64(math.MaxInt64), qbt.quotaBytes)
+
+	// Add more free resources and put a block successfully.
+
+	qbt.updateRemote(10, 100)
+
+	require.Equal(t, int64(1), qbt.unflushedBytes)
+	require.Equal(t, int64(10), qbt.remoteUsedBytes)
+	require.Equal(t, int64(100), qbt.quotaBytes)
+
+	qbt.afterBlockPut(10, true)
+
+	require.Equal(t, int64(11), qbt.unflushedBytes)
+	require.Equal(t, int64(10), qbt.remoteUsedBytes)
+	require.Equal(t, int64(100), qbt.quotaBytes)
+
+	// Then try to put a block but fail it.
+
+	qbt.afterBlockPut(9, false)
+
+	require.Equal(t, int64(11), qbt.unflushedBytes)
+	require.Equal(t, int64(10), qbt.remoteUsedBytes)
+	require.Equal(t, int64(100), qbt.quotaBytes)
+
+	// Finally, flush a block.
+
+	qbt.onBlocksFlush(10)
+
+	require.Equal(t, int64(1), qbt.unflushedBytes)
+	require.Equal(t, int64(10), qbt.remoteUsedBytes)
+	require.Equal(t, int64(100), qbt.quotaBytes)
+}
+
+// TestJournalTrackerCounters checks that a journal tracker's counters
+// are updated properly for each public method.
+func TestJournalTrackerCounters(t *testing.T) {
+	jt, err := newJournalTracker(
+		0.1,  // minThreshold
+		0.9,  // maxThreshold
+		0.8,  // quotaMinThreshold
+		1.2,  // quotaMaxThreshold
+		0.15, // journalFrac
+		400,  // byteLimit
+		800,  // fileLimit
+		100,  // freeBytes
+		200)  // freeFiles
+	require.NoError(t, err)
+
+	byteSnapshot, fileSnapshot := jt.getByteFileSnapshotsForTest()
+	// max = count = min(k(U+F), L) = min(0.15(0+100), 400) = 15.
+	require.Equal(t, jtSnapshot{
+		used:  0,
+		free:  100,
+		max:   15,
+		count: 15,
+	}, byteSnapshot)
+	// max = count = min(k(U+F), L) = min(0.15(0+200), 800) = 30.
+	require.Equal(t, jtSnapshot{
+		used:  0,
+		free:  200,
+		max:   30,
+		count: 30,
+	}, fileSnapshot)
+
+	quotaSnapshot := jt.getQuotaSnapshotForTest()
+	require.Equal(t, jtSnapshot{
+		used: 0,
+		free: math.MaxInt64,
+	}, quotaSnapshot)
+
+	// For stored bytes, increase U by 10, so that increases max
+	// by 0.15*10 = 1.5, so max is now 16. For files, increase U
+	// by 20, so that increases max by 0.15*20 = 3, so max is now
+	// 33.
+
+	availBytes, availFiles := jt.onEnable(10, 5, 20)
+	require.Equal(t, int64(6), availBytes)
+	require.Equal(t, int64(13), availFiles)
+
+	byteSnapshot, fileSnapshot = jt.getByteFileSnapshotsForTest()
+	require.Equal(t, jtSnapshot{
+		used:  10,
+		free:  100,
+		max:   16,
+		count: 6,
+	}, byteSnapshot)
+	require.Equal(t, jtSnapshot{
+		used:  20,
+		free:  200,
+		max:   33,
+		count: 13,
+	}, fileSnapshot)
+
+	quotaSnapshot = jt.getQuotaSnapshotForTest()
+	require.Equal(t, jtSnapshot{
+		used: 5,
+		free: math.MaxInt64 - 5,
+	}, quotaSnapshot)
+
+	// For stored bytes, decrease U by 9, so that decreases max by
+	// 0.15*9 = 1.35, so max is back to 15. For files, decrease U
+	// by 19, so that decreases max by 0.15*19 = 2.85, so max back to
+	// 30.
+
+	jt.onDisable(9, 4, 19)
+
+	byteSnapshot, fileSnapshot = jt.getByteFileSnapshotsForTest()
+	require.Equal(t, jtSnapshot{
```

Comment:
Might make more sense to maintain `expected{Byte,File,Quota}Snapshot` variables here and update them, to avoid having to repeat everything if only one changes.